### PR TITLE
[TargetIntegrator] Only embed application and unit test bundle targets

### LINKED ISSUE #3440: Framework inside embedded framework
I have a project using Xcode 6 with an embedded Framework target. For this target, I have one pod for it and have the use_frameworks! flag set in my Podfile.

When I run `pod install` and then eventually submit to iTunes connect for test flight testing, I get the errors:

* Invalid Bundle. The bundle at 'AppName.app/Frameworks/DataFramework.framework' contains disallowed file 'Frameworks'.
* Invalid Bundle. The bundle at 'AppName.app/Frameworks/DataFramework.framework' contains disallowed texted bundles.

Removing the 'Embed pods Frameworks' and 'Embed Pods Resources' build phases from the framework target. And then adding the scripts that were in these build phases to the corresponding build phases of the main app fixes the issue and everything works fine.

Should pods built into secondary targets have this copying phase only in the main app? Or should the folder they are copied into be called something else? I have noticed other solutions like http://zapone.org/barrie/2015/01/17/validation-error-the-bundle-contains-disallowed-file-frameworks/ that copy the frameworks into a 'Shared Frameworks' folder and the issue goes away.

### PR DESCRIPTION
**Pull request by mrackwitz**
Fixes #3440.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,16 @@ To install or update CocoaPods see this [guide](http://docs.cocoapods.org/guides
 
 To install release candidates run `[sudo] gem install cocoapods --pre`
 
+## Master
+
+##### Bug Fixes
+
+* Only add the "Embed Pods Frameworks" script for application and unit test targets.  
+  [Marius Rackwitz](https://github.com/mrackwitz)
+  [#3489](https://github.com/CocoaPods/CocoaPods/issues/3489)
```
Comment (segiddins) [inline CHANGELOG.md]: don't link the PR

File: lib/cocoapods/installer/user_project_integrator/target_integrator.rb [hunk 2]
```diff
@@ -9,6 +9,11 @@ class UserProjectIntegrator
       class TargetIntegrator
         autoload :XCConfigIntegrator, 'cocoapods/installer/user_project_integrator/target_integrator/xcconfig_integrator'
 
+        # @return [Array<Symbol>] the symbol types, which require that the pod
+        # frameworks are embedded in the output directory / product bundle.
+        #
+        EMBED_FOR_SYMBOL_TYPES = [:application, :unit_test_bundle].freeze
```
Comment (segiddins) [inline lib/cocoapods/installer/user_project_integrator/target_integrator.rb]: `EMBED_FRAMEWORK_TARGET_TYPES`

File: lib/cocoapods/installer/user_project_integrator/target_integrator.rb [hunk 3]
```diff
@@ -136,7 +141,10 @@ def add_pods_library
         #
         def add_embed_frameworks_script_phase
           phase_name = 'Embed Pods Frameworks'
-          native_targets_to_integrate.each do |native_target|
+          targets_which_embed = native_targets_to_integrate.select do |target|
```
Comment (segiddins) [inline lib/cocoapods/installer/user_project_integrator/target_integrator.rb]: `targets_to_embed_in`

**Comment by mrackwitz** [comment]
@segiddins: Thanks for the feedback.

File: spec/unit/installer/user_project_integrator/target_integrator_spec.rb [hunk 4]
```diff
@@ -115,6 +115,25 @@ module Pod
           phase.nil?.should == true
         end
 
+        it 'does not add an embed frameworks build phase if the target to integrate is a framework' do
+          target = @target_integrator.send(:native_targets).first
+          target.stubs(:symbol_type).returns(:framework)
+          @target_integrator.integrate!
+          phase = target.shell_script_build_phases.find { |bp| bp.name == 'Embed Pods Frameworks' }
+          phase.nil?.should == true
```
Comment (segiddins) [inline spec/unit/installer/user_project_integrator/target_integrator_spec.rb]: `phase.should.be.nil` ?

**Comment by segiddins** [comment]
:+1: seems good

File: spec/unit/installer/user_project_integrator/target_integrator_spec.rb (hunk 4 above)
Comment (mrackwitz) [inline spec/unit/installer/user_project_integrator/target_integrator_spec.rb]: `that.seems.legit.to_me` :+1:

**Comment by neonichu** [comment]
LGTM :+1:

**Comment by damienpontifex** [comment]
Sorry for my inability to decipher the full extent of this. Just making sure...

I see it will only add the script for application and unit test bundles, but will it still embed frameworks that are referenced only against a Framework target?
i.e. pods linked against a Dynamic Framework, I don't see that they will be in the install_framework list of the build script

**Comment by pixel4** [comment]
Hey guys,

I don't know a lot about this process in Cocoapods but seems like this change is causing my `UITests` target to NOT get the Build Phase "📦 Embed Pods Frameworks".

Anyone else having this issue?

**Comment by neonichu** [comment]
We're aware of that issue, see #5250

**Pull request merged**